File: medtronic/src/main/java/info/nightscout/androidaps/plugins/pump/medtronic/data/dto/RLHistoryItemMedtronic.java
package info.nightscout.androidaps.plugins.pump.medtronic.data.dto;

import org.joda.time.LocalDateTime;

import info.nightscout.androidaps.plugins.pump.common.hw.rileylink.data.RLHistoryItem;
import info.nightscout.androidaps.plugins.pump.common.hw.rileylink.defs.RileyLinkTargetDevice;
import info.nightscout.androidaps.plugins.pump.medtronic.defs.MedtronicCommandType;
import info.nightscout.androidaps.utils.resources.ResourceHelper;

public class RLHistoryItemMedtronic extends RLHistoryItem {

    private final MedtronicCommandType medtronicCommandType;

    public RLHistoryItemMedtronic(MedtronicCommandType medtronicCommandType) {
        super(new LocalDateTime(), RLHistoryItemSource.MedtronicCommand, RileyLinkTargetDevice.MedtronicPump);
        this.medtronicCommandType = medtronicCommandType;
    }

    @Override
    public String getDescription(ResourceHelper resourceHelper) {
        if (RLHistoryItemSource.MedtronicCommand.equals(source)) {
            return medtronicCommandType.name();
        }

        return super.getDescription(resourceHelper);
    }

}

File: pump/medtronic/src/main/kotlin/app/aaps/pump/medtronic/data/dto/RLHistoryItemMedtronic.kt
package app.aaps.pump.medtronic.data.dto

import app.aaps.core.interfaces.resources.ResourceHelper
import app.aaps.pump.common.hw.rileylink.data.RLHistoryItem
import app.aaps.pump.common.hw.rileylink.defs.RileyLinkTargetDevice
import app.aaps.pump.medtronic.defs.MedtronicCommandType
import org.joda.time.LocalDateTime

class RLHistoryItemMedtronic(private val medtronicCommandType: MedtronicCommandType) :
    RLHistoryItem(LocalDateTime(), RLHistoryItemSource.MedtronicCommand, RileyLinkTargetDevice.MedtronicPump) {

    override fun getDescription(rh: ResourceHelper): String =
        if (RLHistoryItemSource.MedtronicCommand == source) medtronicCommandType.name
        else super.getDescription(rh)
}